sii/resource.py

```diff
@@ -41,16 +41,16 @@ def get_iva_values(invoice, in_invoice):
             else:
                 sign = SIGN[invoice.rectificative_type]
                 iva = {
-                    'BaseImponible': sign * inv_tax.base,
+                    'BaseImponible': sign * abs(inv_tax.base),
                     'TipoImpositivo': inv_tax.tax_id.amount * 100
                 }
                 if in_invoice:
-                    iva['CuotaRepercutida'] = sign * inv_tax.tax_amount
+                    iva['CuotaRepercutida'] = sign * abs(inv_tax.tax_amount)
                 else:
-                    iva['CuotaSoportada'] = sign * inv_tax.tax_amount
+                    iva['CuotaSoportada'] = sign * abs(inv_tax.tax_amount)
                 vals['iva_no_exento'] = True
                 vals['detalle_iva'].append(iva)
-        else:
+        elif 'Impuesto especial sobre la electricidad' != inv_tax.name.lower():
```

Comment:
No té perquè sempre el text ser així, hi ha uns 'Impostos especials sobre l'electricitat' que tenen una part exempta potser pots fer un `'sobre la electricidad' in inv_tax.name.lower()`